src/test/java/org/jenkinsci/plugins/workflow/support/pickles/serialization/DeserializeUpdate.java

```diff
@@ -0,0 +1,44 @@
+package org.jenkinsci.plugins.workflow.support.pickles.serialization;
+
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputAction;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputStepExecution;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.util.List;
+
+/**
+ * Tests deserializing program.dat, useful to catch
+ */
+public class DeserializeUpdate {
```

Comment:
Makes sense to upstream the test to the master branch. It is helpful independently of whether we merge it or not

Reply:
We don't generally upstream failing tests

Reply:
But it should be passing on Java 8, no?

Reply:
@oleg-nenashev No, because the actual code binary format is different, on Java 8 as well -- it's the bump to the marshaller that does it, not the specific java version in use.